internal/sign/sign.go

```diff
@@ -0,0 +1,43 @@
+package sign
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+)
+
+func GetHmacString(body []byte, key string) string {
+	if key == "" {
+		return ""
+	}
+	h := hmac.New(sha256.New, []byte(key))
+	_, err := h.Write(body)
+	if err != nil {
+		return ""
+	}
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func Verify(msg interface{}, key, hashString string) bool {
+	if key == "" || hashString == "" {
```

Comment:
```suggestion
	if key == "" {
```
Если в сервере включена проверка, то неподписанные запросы он должен отлуплять.

Reply:
да, я это понимаю, но вот есть в тестах момент, где отправляется запрос на апдейт метрик на сервер, который запущен с ключом, но запрос отправлен без хедера `HashSHA256` c соответствующим вычисленным ключом. https://github.com/Yandex-Practicum/go-autotests/blob/main/cmd/metricstest/iteration14_test.go#L267
Таким образом, тесты последние падают.

Reply:
Причина, по которой я добавляла эту строку я привела проконсолив вот в это [жобе](https://github.com/aykuli/observer/actions/runs/9386824207/job/25848307025)

тут как будто агент не получает ключ шифрования, а сервер получает - вот и асинхрон получается.

![Screenshot from 2024-06-05 22-48-56](https://github.com/aykuli/observer/assets/49780822/254a9784-b5de-42bd-b1da-128b5f81cf34)